contrib/babelfishpg_tds/src/include/tds_request.h

```diff
@@ -536,21 +666,60 @@ SetColMetadataForTvp(ParameterToken temp, const StringInfo message, uint64_t *of
 			if (i == 1)
 				temp->tvpInfo->tvpTypeSchemaName = tempStringInfo->data;
 			else
+			{
 				temp->tvpInfo->tvpTypeName = tempStringInfo->data;
-
+				temp->tvpInfo->tableName = tempStringInfo->data;
+			}
 		}
 		else if (i == 2)
 		{
-			/* Throw error if TabelType-Name is not provided */
-			ereport(ERROR,
-					(errcode(ERRCODE_PROTOCOL_VIOLATION),
-					 errmsg("The incoming tabular data stream (TDS) remote procedure call (RPC) protocol stream is incorrect. "
-							"Table-valued parameter %d, to a parameterized string has no table type defined.",
-							temp->paramOrdinal + 1)));
+			if (!xactStarted)
+				StartTransactionCommand(); 
+			user_id = GetSessionUserId();
```

Comment:
This is incase of cross database, we might need to use GetUserId. Please check with harsh

Reply:
As currently we don't support cross-db for TVP proc, changed it to GetUserId